ui/desktop/src/components/ChatInput.tsx

```diff
@@ -572,6 +572,7 @@ export default function ChatInput({
         },
         compactIcon: <ScrollText size={12} />,
         autoCompactThreshold: autoCompactThreshold,
+        key: `context-${numTokens}-${tokenLimit}`, // Force re-render on token count change
```

Comment:
this doesn't strike me as the correct solution, at most a work around? why are we not rerendering this when the numTokens changes? it is in the alert properties. do we do a shallow compare somewhere?